orchagent/portsorch.cpp

```diff
@@ -345,6 +350,132 @@ bool PortsOrch::setPortMtu(sai_object_id_t id, sai_uint32_t mtu)
     return true;
 }
 
+bool PortsOrch::setPortPvid (Port &port, sai_uint32_t pvid)
+{
+    SWSS_LOG_ENTER();
+
+    vector<sai_object_id_t>   idv;
+
+    if (port.m_type == Port::PHY)
+    {
+        idv.push_back(port.m_port_id);
+    }
+    else if (port.m_type == Port::LAG)
+    {
+        Port member;
+        auto it = port.m_members.begin();
+         while (it != port.m_members.end())
+         {
+            const string& memberName = *it;
+            if (!getPort(memberName, member))
+            {
+                SWSS_LOG_ERROR("Failed to get port for %s alias\n", memberName.c_str());
+                return false;
+            }
+            idv.push_back(member.m_port_id);
+            it++;
+         }
+    }
+    else
+    {
+        SWSS_LOG_ERROR("PortsOrch::setPortPvid port type %d not supported", port.m_type);
+        return false;
+    }
+
+    for (const auto& id: idv)
+    {
+        sai_attribute_t attr;
+        attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
+        attr.value.u32 = pvid;
+
+        sai_status_t status = sai_port_api->set_port_attribute(id, &attr);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to set pvid %u to port pid:%lx", attr.value.u32, id);
+            return false;
+        }
+        SWSS_LOG_INFO("Set pvid %u to port pid:%lx", attr.value.u32, id);
+    }
+    return true;
+}
+
+bool PortsOrch::getPortPvid(Port &port, sai_uint32_t &pvid)
+{
+
+    if (port.m_port_vlan_id != DEFAULT_PORT_VLAN_ID)
+    {
+        pvid = port.m_port_vlan_id;
+        return true;
+    }
+
+    if (port.m_vlan_members.size() == 0)
+    {
+        pvid = DEFAULT_PORT_VLAN_ID;
+        return true;
+    }
+
+    for (auto &vme: port.m_vlan_members)
+    {
+        if(vme.second.vlan_mode == SAI_VLAN_TAGGING_MODE_UNTAGGED) {
```

Comment:
how to ensure only one untagged mode is applied to the port's VLANs?
we could initialize the port with the default port VLAN ID.
as the logic below already set the port VLAN ID when the VLAN is untagged, here we could directly return `port.m_port_vlan_id` instantly.

Reply:
No control of number of untagged mode in orchagent, config manager is supposed to prevent port from being put into multiple VLAN as untagged.   

m_port_vlan_id is only set upon explicit pvid configuration.  It has precedence over the implicit pvid from untagged VLAN.

Reply:
here the port VLAN ID is valid only when we are querying the physical port, is that right? a physical port will not have VLAN members. I'm not sure about why the `for` loop is here. Do we need to add checks at the entrance of this function to ensure the type of the port? If a sync is needed, let's discuss this on the phone.

Reply:
here m_vlan_members stores all the VLANs that the port has joined in tagged/untagged mode.   Ex. port Ethernet41 joined VLAN 100, 200, and 300, we loop through all the VLANs and find the one in which Ethernet41 is of untagged mode